src/InsetOrderOptimizer.cpp

```diff
@@ -99,127 +107,234 @@ bool InsetOrderOptimizer::optimize(const WallType& wall_type)
+        pack_by_inset?
+        getInsetOrder(walls_to_be_added, outer_to_inner)
+        : getRegionOrder(walls_to_be_added, outer_to_inner);
+    
+    if (center_last)
+    {
+        for (const ExtrusionLine* line : walls_to_be_added)
+            if (line->is_odd)
+                for (const ExtrusionLine* other_line : walls_to_be_added)
+                    if ( ! other_line->is_odd)
+                        order.emplace(std::make_pair(other_line, line));
+    }
+    
+    constexpr Ratio flow = 1.0_r;
+    
+    bool added_something = false;
 
-        const bool alternate_direction_modifier = inset % 2 == 0;
-        if(bins_with_index_zero_insets.count(inset) > 0) //Print using outer wall config.
+    constexpr bool detect_loops = false;
+    constexpr Polygons* combing_boundary = nullptr;
+    //When we alternate walls, also alternate the direction at which the first wall starts in.
+    //On even layers we start with normal direction, on odd layers with inverted direction.
+    constexpr bool reverse_all_paths = false;
+    PathOrderOptimizer<const ExtrusionLine*> order_optimizer(gcode_layer.getLastPlannedPositionOrStartingPosition(), z_seam_config, detect_loops, combing_boundary, reverse_all_paths, order);
+    
+    for (const ExtrusionLine* line : walls_to_be_added)
+    {
+        if (line->is_closed)
         {
-            gcode_layer.addWalls(insets[inset], mesh.settings, inset_0_non_bridge_config, inset_0_bridge_config, z_seam_config, wall_0_wipe_dist, flow, retract_before_outer_wall, alternate_direction_modifier);
+            order_optimizer.addPolygon(line);
         }
         else
         {
-            gcode_layer.addWalls(insets[inset], mesh.settings, inset_X_non_bridge_config, inset_X_bridge_config, z_seam_config, 0, flow, false, alternate_direction_modifier);
+            order_optimizer.addPolyline(line);
         }
     }
+    
+    
+    order_optimizer.optimize();
+    
+    cura::Point p_end {0, 0};
+    for(const PathOrderOptimizer<const ExtrusionLine*>::Path& path : order_optimizer.paths)
+    {
+        if (path.vertices->empty()) continue;
+        
+        const bool is_outer_wall = path.vertices->inset_idx == 0; // or thin wall 'gap filler'
+        const bool is_gap_filler = path.vertices->is_odd;
+        const GCodePathConfig& non_bridge_config = is_outer_wall ? inset_0_non_bridge_config : inset_X_non_bridge_config;
+        const GCodePathConfig& bridge_config = is_outer_wall? inset_0_bridge_config : inset_X_bridge_config;
+        const coord_t wipe_dist = is_outer_wall && ! is_gap_filler ? wall_0_wipe_dist : wall_x_wipe_dist;
+        const bool retract_before = is_outer_wall ? retract_before_outer_wall : false;
+        
+        const bool alternate_direction_modifier = alternate_walls && (path.vertices->inset_idx % 2 == layer_nr % 2);
+        const bool backwards = path.backwards != alternate_direction_modifier;
```

Comment:
`^` is a bitwise operator, whereas `!=` is the xor for boolean values. Now for boolean values there is only one bit of interest, so it would work, but idk - it seems like the inappropriate operator. We also wouldn't use the bitwise `&` or the bitwise `|`.

I read up on it and it turns out we also shouldn't be using the bitwise assignments `&=` and `|=` for that matter.

Reply:
I've applied your changes, bit with `!=` instead of `^`.

Reply:
You could also take the sum of all these boolean values and do a `mod 2` on the result, doesn't increase readability though.

Reply:
Lol